### ISSUE TITLE
Incorrect alignment of ruby, and incorrect tone colors

### ISSUE THREAD

**Issue by onaphta** [luoliyan/chinese-support-redux#32]
**Describe the bug**
I have noticed some problems with zhuyin and then ruby when creating new cards. 
The zhuyin is right but the color of the first character is always that of the second. For exemple, ㄐㄧㄔㄤˇ appears all in green.
The second character is often missing, but the whole zhuyin has the colour of the second character. 
For instance, for 機場 in the ruby field I get 機 and above ㄐㄧㄔㄤˇ in green.

However, if I add manually a space between characters in the zhuyin field both the zhuyin color and the ruby field will become correct. The pinyin field and the colour field will remain wrong.

I guess the problem has to do with that space?

**Specs (please complete the following information):**
 - OS: Windows 10
 - Anki Version 2.1.7
- Chinese Support Version 0.90

**Comment by luoliyan** [comment]
This is probably fallout from an internal change I've been slowly making to how sentences are handled. Previously the add-on aggressively separated syllables and ignored word boundaries. My strategy has been to fix this by passing sentences around as a list of words, but we are still in the transition period and it's not pretty, so we get issues like this.

I'll try to fix this one up tonight.

**Comment by mawis** [comment]
I had a somehow similar experience right now. The word I typed in was 空虚 (kong1xu1, ㄎㄨㄥ ㄒㄩ).

The plugin generated ㄎㄨㄥˋ ㄒㄩ (including a space between the syllables. Coloring and Ruby was as expected for the fourth tone. I removed the falling tone mark (keeping the space between the syllables). Then the coloring was both syllables first tone (correct), but the ruby field lost the second syllable and had both Zhuyin syllables annotated to only the first character.

Anki: 2.1.4
Chinese Support Redux: 0.9.1 (+ the patch I submitted today)

**Comment by mawis** [comment]
I just noted this additional fact: After I put **two** spaces between the syllables in the ㄅㄆㄇㄈ field I get the correct content in the RubyBPMF field (two syllables, both annotated, both in the color of the first tone). But now the coloring in the ㄅㄆㄇㄈ field is incorrect again.

**Comment by EmilRehnberg** [comment]
I'm probably being redundant here but unexpected behaviour seems to crop up with e.g. "就是没香气". Here the pinyin for 就是 all lands in 就 and the rest of the ruby is out of sync. 
When using Google Translate 就是 is grouped together in pinyin so that's maybe related. 

I'm still on '2019.01.17' release since that's what's up on ankiweb.net.

**Comment by JasonBens** [comment]
Yeah.  I've got a fork that I spent some time debugging the pinyin and ruby output on.  Part of the problem seems to be that the regex to detect pinyin doesn't capture some of the characters.  Characters without an initial consonant, I think, come to mind.  This makes other things break, and then bad things happen.  I'm not a developer on this project, but I may have a pull request someday when offline commitments calm down enough that I can address this.
